tests/regression/src/lib.rs

```diff
@@ -1,48 +1,72 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-use s2n_tls::{
-    config::Builder,
-    security,
-    testing::{CertKeyPair, InsecureAcceptAllCertificatesHandler},
-};
-type Error = s2n_tls::error::Error;
-
-/// Function to create default config with specified parameters.
-pub fn set_config(
-    cipher_prefs: &security::Policy,
-    keypair: CertKeyPair,
-) -> Result<s2n_tls::config::Config, Error> {
-    let mut builder = Builder::new();
-    builder
-        .set_security_policy(cipher_prefs)
-        .expect("Unable to set config cipher preferences");
-    builder
-        .set_verify_host_callback(InsecureAcceptAllCertificatesHandler {})
-        .expect("Unable to set a host verify callback.");
-    builder
-        .load_pem(keypair.cert(), keypair.key())
-        .expect("Unable to load cert/pem");
-    builder.trust_pem(keypair.cert()).expect("load cert pem");
-    builder.build()
+pub mod git {
+    use std::process::Command;
+
+    pub fn get_current_commit_hash() -> String {
+        let output = Command::new("git")
+            .args(["rev-parse", "HEAD"])
+            .output()
+            .expect("Failed to get commit hash");
+        assert!(output.status.success());
+
+        String::from_utf8(output.stdout)
+            .expect("Invalid UTF-8 in commit hash")
+            .trim()
+            .to_string()
+    }
+
+    pub fn is_commit_in_log(commit: &str) -> bool {
+        let output = Command::new("git")
+            .args(["log", "--pretty=format:%H"])
+            .output()
+            .expect("Failed to execute git log");
+        assert!(output.status.success());
+
+        let log = String::from_utf8(output.stdout).expect("Invalid UTF-8 in git log output");
+        log.lines().any(|line| line == commit)
```

Comment:
is this going to go through the entire commit log? I guess it's currently sitting at 4814 commits but can you not use `merge-base` for this?

Reply:
You're right, we can use merge-base for this. Since I already use merge-base to figure out which commit is older, I can just add a condition which will panic if neither commit is resolved as older, meaning the commit isn't in the log: 
`if git::is_older_commit(&profile1.commit_hash, &profile2.commit_hash) {
    (profile1, profile2)
} else if git::is_older_commit(&profile2.commit_hash, &profile1.commit_hash) {
    (profile2, profile1)
} else {
    panic!("The commit is not in the log");
}`